### PR TITLE
refactor: migrate TextPaddlehubEncoder

### LINKED ISSUE #166: migrate TextPaddlehubEncoder
┆Issue is synchronized with this [Jira Task](https://jinaai.atlassian.net/browse/JINAHUB-46) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by nan-wang**
resolved #166

### CODE REVIEW

File: encoders/nlp/TextPaddlehubEncoder/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM paddlepaddle/paddle
+
+# setup the workspace
+COPY . /workspace
+WORKDIR /workspace
+
+# install the third-party requirements
+# RUN pip install -r requirements.txt
+RUN pip install jina
```
Comment (JoanFM) [inline encoders/nlp/TextPaddlehubEncoder/Dockerfile:9]: `jina[devel]`?

File: encoders/nlp/TextPaddlehubEncoder/tests/test_textpaddlehubencoder.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    def __init__(self):
+        print('i am a mocker module')
+
+    def get_embedding(self, texts, *args, **kwargs):
+        print('i am a mocker embedding')
+        return [[np.random.random(target_output_dim), None]] * len(texts)
+
+
+@mock.patch('paddlehub.Module', return_value=MockModule())
+def test_textpaddlehubencoder_encode(mocker):
```
Comment (JoanFM) [inline encoders/nlp/TextPaddlehubEncoder/tests/test_textpaddlehubencoder.py:39]: is there a chance to add an extra test with a real model?

File: encoders/nlp/TextPaddlehubEncoder/Dockerfile (hunk 1 above)
Comment (nan-wang) [reply encoders/nlp/TextPaddlehubEncoder/Dockerfile:9]: switched back to the jina base image. The `paddlepaddle/paddle` is more than 10G!

File: encoders/nlp/TextPaddlehubEncoder/tests/test_textpaddlehubencoder.py (hunk 2 above)
Comment (nan-wang) [reply encoders/nlp/TextPaddlehubEncoder/tests/test_textpaddlehubencoder.py:39]: Yes, I plan to set two types of test. One is for the cicd process, which use mocks for testing. The other is for building the Docker image, in which the real model will be loaded.

File: encoders/nlp/TextPaddlehubEncoder/Dockerfile [hunk 3]
```diff
@@ -1,14 +1,13 @@
-FROM paddlepaddle/paddle
+FROM jinaai/jina
```
Comment (JoanFM) [inline encoders/nlp/TextPaddlehubEncoder/Dockerfile:1]: should we use `devel`?

Comment (JoanFM) [reply encoders/nlp/TextPaddlehubEncoder/Dockerfile]: sorry apparently the new cookiecutter puts `jinaai/jina` without devel so I may be wrong

File: encoders/nlp/TextPaddlehubEncoder/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+WORKDIR /workspace
+
+RUN apt-get update && \
+    apt-get -y install libgomp1 libgl1-mesa-glx libglib2.0-0
+
+# install the third-party requirements
+RUN pip install -r requirements.txt && pip uninstall -y pathlib
+
+# for testing the image
+RUN pip install -y pytest pytest-mock mock && JINA_TEST_PRETRAINED='true' pytest
```
Comment (rutujasurve94) [inline encoders/nlp/TextPaddlehubEncoder/Dockerfile:14]: ```&& EXPORT JINA_TEST_PRETRAINED='true'``` instead of ```JINA_TEST_PRETRAINED='true'```

Comment (JoanFM) [reply encoders/nlp/TextPaddlehubEncoder/Dockerfile:14]: it actually seems to work without export

Comment (nan-wang) [reply encoders/nlp/TextPaddlehubEncoder/Dockerfile]: - `-y` is necessary for uninstallation.
- `EXPORT` is not necessary

**Pull request merged**